### PR TITLE
CORS added for cross-site requests and readme updated

### PR DESCRIPTION
**Pull request by simonpoole**
This is based on https://github.com/komoot/photon/pull/373 and preserves rahulhaque commits but simplifies things a tiny bit and rebases it on the current master.

Currently the -cors flag is all or nothing, but I would expect that people running photon that want more fine grained control would put it behind a reverse proxy in any case.

It doesn't resolve the mystery of why the previous code that added a Access-Control-Allow-Origin header seems to have worked for some (for example myself) and not for others.

### CODE REVIEW

File: src/main/java/de/komoot/photon/App.java [hunk 1]
```diff
@@ -134,9 +135,17 @@ private static void startNominatimImport(CommandLineArgs args, Server esServer,
      * @param esNodeClient
      */
     private static void startApi(CommandLineArgs args, Client esNodeClient) {
-        setPort(args.getListenPort());
-        setIpAddress(args.getListenIp());
-
+        port(args.getListenPort());
+        ipAddress(args.getListenIp());
```
Comment (lonvia) [inline src/main/java/de/komoot/photon/App.java:142]: This looks unrelated. Any particular reason to change it?

**Comment by lonvia** [comment]
I suspect that the difference if the old version works or not is if the caller does an OPTION request first.

File: src/main/java/de/komoot/photon/App.java (hunk 1 above)
Comment (simonpoole) [reply src/main/java/de/komoot/photon/App.java:142]: The old versions are deprecated in spark but functionally exactly the same, changed here to get rid of the warnings.

I'll create a separate PR if you promise to merge it :-)

**Comment by simonpoole** [comment]
As a tendency I would prefer -cors-all without an argument and -cors (or -cors-origin) with one.

**Comment by simonpoole** [comment]
> 
> 
> I suspect that the difference if the old version works or not is if the caller does an OPTION request first.

I assume so too, I just couldn't recreate with any of the (very mainstream and current) browsers I use.

**Comment by karussell** [comment]
This looks good. (Although I would not expose photon to the public and instead use nginx or similar tools to proxy it for unrelated security reasons)

**Comment by prasadrajuv** [comment]
add https (ssl) support to avoid Blocked loading mixed active content in browser because geolocation support removed for http protocol
